packages/frontend/src/features/sqlRunner/store/cartesianChartBaseSlice.ts

```diff
@@ -0,0 +1,195 @@
+import {
+    DEFAULT_AGGREGATION,
+    type AggregationOptions,
+    type BarChartSqlConfig,
+    type CartesianChartDisplay,
+    type GroupByLayoutOptions,
+    type LineChartSqlConfig,
+    type SqlTransformCartesianChartConfig,
+    type XLayoutOptions,
+    type YLayoutOptions,
+} from '@lightdash/common';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { createSlice } from '@reduxjs/toolkit';
+
+type InitialState = {
+    defaultLayout: SqlTransformCartesianChartConfig | undefined;
+    config: BarChartSqlConfig | LineChartSqlConfig | undefined;
+    options: {
+        xLayoutOptions: XLayoutOptions[];
+        yLayoutOptions: YLayoutOptions[];
+        groupByOptions: GroupByLayoutOptions[];
+    };
+};
+
+const initialState: InitialState = {
+    defaultLayout: undefined,
+    config: undefined,
+    options: {
+        xLayoutOptions: [],
+        yLayoutOptions: [],
+        groupByOptions: [],
+    },
+};
+
+export const cartesianChartConfigSlice = createSlice({
+    name: 'cartesianChartConfig',
```

Comment:
should we rename this to have Base to match the file name?

Reply:
Sure. I'm not sure this does anything at all but might as well match.